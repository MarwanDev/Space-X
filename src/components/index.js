export { default as Navbar } from './Navbar/Navbar';
export { default as Rockets } from './Rockets/Rockets';
export { default as Missions } from './Missions/Missions';
export { default as MyProfile } from './MyProfile/MyProfile';

