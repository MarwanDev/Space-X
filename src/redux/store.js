import { configureStore } from "@reduxjs/toolkit";
import rocketReducer from "./RocketSlice/rocketsSlice";


const store = configureStore({
  reducer: {
    rockets: rocketReducer,
  },

});

export default store;
