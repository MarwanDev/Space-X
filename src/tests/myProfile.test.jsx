import { expect, it, describe } from "vitest";
import renderer from "react-test-renderer";
import store from "../redux/store";
import { Provider } from "react-redux";
import { MyProfile } from "../components";

describe("MyProfile", () => {
  it("renders MyProfile correctly", () => {
    const myProfile = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>
      )
      .toJSON();
    expect(myProfile).toMatchSnapshot();
  });
});
