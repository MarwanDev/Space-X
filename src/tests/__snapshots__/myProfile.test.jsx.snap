// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyProfile > renders MyProfile correctly 1`] = `
<div
  style={
    {
      "display": "flex",
      "paddingLeft": 90,
      "paddingRight": 90,
    }
  }
>
  <div
    className="container-md"
  >
    <div
      className="p-3"
    >
      <h2>
        My Missions
      </h2>
    </div>
  </div>
  <div
    className="container-md"
  >
    <div
      className="p-3"
    >
      <h2>
        My Rockets
      </h2>
    </div>
  </div>
</div>
`;
